### PR TITLE
fix: Changed inplace spike latency encode to non-inplace

### PR DESCRIPTION
**Pull request by erikberter**
The functions spike_latency_lif_encode and spike_latency_encode in the functional modulo have in-place operations while applying the mask. I replace the boolean mask with an integer mask which tracks the aggregate of the spikes, which can be used with a comparator inside the torch.where.

I found the error while trying to build a model with an analogic convolutional neural network to extract features before a spiking classificator with time-to-first spike encoding. After the change was made, the video classificator worked fine.

### CODE REVIEW

**Comment by cpehle** [comment]
Thanks, this looks great. We would be happy if you could share the project you are working at some point!

**Review by cpehle — APPROVED** [review]
Thanks, this does enable flowing gradients through this layer aswell right?

**Comment by erikberter** [comment]
> this does enable flowing gradients through this layer aswell right?

Indeed, I was able to fine-tune a ResNet-3D-18 through it.
